lib/services/platform-service.ts

```diff
@@ -458,16 +458,25 @@ export class PlatformService implements IPlatformService {
 		if (this.$options.availableDevices) {
 			return this.$emulatorPlatformService.listAvailableEmulators(platform);
 		}
+		this.$options.emulator = true;
 		if (this.$options.device) {
 			let info = this.$emulatorPlatformService.getEmulatorInfo(platform, this.$options.device).wait();
 			if (info) {
 				if (!info.isRunning) {
 					this.$emulatorPlatformService.startEmulator(info).wait();
 				}
 				this.$options.device = null;
+			} else {
+				this.$devicesService.initialize({ platform: platform, deviceId: this.$options.device }).wait();
+				let found: Mobile.IDeviceInfo[] = [];
+				if (this.$devicesService.hasDevices) {
```

Comment:
Why is this code required? `devicesService` will fail in case you pass `--device invalidIdentifier` and it does not find such identifier:
https://github.com/telerik/mobile-cli-lib/blob/master/mobile/mobile-core/devices-service.ts#L337
https://github.com/telerik/mobile-cli-lib/blob/master/mobile/mobile-core/devices-service.ts#L345
https://github.com/telerik/mobile-cli-lib/blob/master/mobile/mobile-core/devices-service.ts#L256